internal/services/gcp_reservation_service.go

```diff
@@ -0,0 +1,121 @@
+package services
+
+import (
+	"net/http"
+
+	"github.com/aws/smithy-go/ptr"
+	"github.com/go-chi/render"
+	"github.com/lzap/dejq"
+
+	"github.com/RHEnVision/provisioning-backend/internal/clients"
+	"github.com/RHEnVision/provisioning-backend/internal/ctxval"
+	"github.com/RHEnVision/provisioning-backend/internal/dao"
+	"github.com/RHEnVision/provisioning-backend/internal/jobs"
+	"github.com/RHEnVision/provisioning-backend/internal/jobs/queue"
+	"github.com/RHEnVision/provisioning-backend/internal/models"
+	"github.com/RHEnVision/provisioning-backend/internal/payloads"
+)
+
+func CreateGCPReservation(w http.ResponseWriter, r *http.Request) {
+	logger := ctxval.Logger(r.Context())
+
+	var accountId int64 = ctxval.AccountId(r.Context())
+
+	// TODO: meanwhile these values are hardcoded until we will have instance types, sources, ssh endpoints,
+	payload := &payloads.GCPReservationRequestPayload{
+		// TODO: The actual value of the pubkey change to pubkey id
+		PubkeyID: "aabramov:ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQC8w6DONv1qn3IdgxSpkYOClq7oe7davWFqKVHPbLoS6+dFInru7gdEO5byhTih6+PwRhHv/b1I+Mtt5MDZ8Sv7XFYpX/3P/u5zQiy1PkMSFSz0brRRUfEQxhXLW97FJa7l+bej2HJDt7f9Gvcj+d/fNWC9Z58/GX11kWk4SIXaKotkN+kWn54xGGS7Zvtm86fP59Srt6wlklSsG8mZBF7jVUjyhAgm/V5gDFb2/6jfiwSb2HyJ9/NbhLkWNdwrvpdGZqQlYhnwTfEZdpwizW/Mj3MxP5O31HN45aE0wog0UeWY4gvTl4Ogb6kescizAM6pCff3RBslbFxLdOO7cR17 lzap+rsakey+{{$randomInt}}@redhat.com",
+		// TODO: The project id change to source id from sources
+		SourceID:    "citric-expanse-361512",
+		Name:        ptr.String("instance1"),
+		Zone:        "us-central1-a",
+		MachineType: "n1-standard-1",
+		ImageID:     "bc97177c-9d07-4db9-ad59-8b2c0bf4174e",
+		PowerOff:    true,
+	}
+	// TODO: Uncomment after removing hard-coded values
+	// if err := render.Bind(r, payload); err != nil {
+	// 	renderError(w, r, payloads.NewInvalidRequestError(r.Context(), err))
+	// 	return
+	// }
+
+	rDao, err := dao.GetReservationDao(r.Context())
+	if err != nil {
+		renderError(w, r, payloads.NewInitializeDAOError(r.Context(), "reservation DAO", err))
+		return
+	}
+
+	detail := &models.GCPDetail{
+		Name:        payload.Name,
+		Zone:        payload.Zone,
+		MachineType: payload.MachineType,
+		Amount:      2,
+		PowerOff:    payload.PowerOff,
+	}
+	reservation := &models.GCPReservation{
+		// TODO: The actual value of the pubkey change to pubkey id
+		PubkeyID: payload.PubkeyID,
+		// TODO: The project id change to source id from sources
+		ImageID:  payload.ImageID,
+		SourceID: payload.SourceID,
+		Detail:   detail,
+	}
+
+	reservation.AccountID = accountId
+	reservation.Status = "Created"
+	reservation.Provider = models.ProviderTypeGCP
+	reservation.Steps = 2
```

Comment:
Isn't this just a single step?

```suggestion
	reservation.Steps = 1
```

Reply:
This was copied from aws, so maybe we need to change it there as well

Reply:
We have pubkey step in AWS, we do not have that yet here :)

Reply:
@lzap this is very hard to remember to update, when adding another step.

Reply:
Understood, it is informative only luckily. But we should find a better solution, agreed.